### PR TITLE
Add support for Microsoft Defender for containers

### PR DESCRIPTION
**Pull request by landerss1**
This PR adds support for Microsoft Defender for containers as an alternative to Trivy. It also adds the option of sending Kubernetes audit logs to an analytics workspace where it can be consumed by Sentinel.

### CODE REVIEW

File: modules/azure/aks/defender.tf [hunk 1]
```diff
@@ -0,0 +1,117 @@
+resource "azurerm_security_center_auto_provisioning" "this" {
+  auto_provision = "Off"
+}
+
+resource "azurerm_security_center_subscription_pricing" "containers" {
+  count         = var.defender_enabled ? 1 : 0
```
Comment (phillebaba) [inline modules/azure/aks/defender.tf:6]: Use for each instead of counts.

File: modules/azure/aks/variables.tf [hunk 2]
```diff
@@ -205,6 +205,47 @@ variable "log_eventhub_authorization_rule_id" {
...
+  default     = false
+}
+
+variable "audit_config" {
+  description = "Kubernetes audit log configuration"
+  type = object({
+    destination_type = optional(string, "StorageAccount")
+    analytics_workspace = optional(object({
+      sku_name       = optional(string, "PerGB2018")
+      daily_quota_gb = optional(number, -1)
```
Comment (phillebaba) [inline modules/azure/aks/variables.tf:220]: Does -1 mean infinite?

Comment (landerss1) [reply modules/azure/aks/variables.tf:220]: Yes. Do we have a better number to use?

File: modules/azure/aks/defender.tf (hunk 1 above)
Comment (landerss1) [reply modules/azure/aks/defender.tf:6]: What is the advantage?

File: modules/azure/aks/variables.tf (hunk 2 above)
Comment (phillebaba) [reply modules/azure/aks/variables.tf:220]: I just mean does it also mean infinite for the API.

File: modules/azure/aks/defender.tf (hunk 1 above)
Comment (phillebaba) [reply modules/azure/aks/defender.tf:6]: This is one of the few cases where count could be accepted, but we have been avoiding it for a long time now. So its just a design choice.

File: modules/azure/aks/variables.tf (hunk 2 above)
Comment (landerss1) [reply modules/azure/aks/variables.tf:220]: It means "The workspace daily quota for ingestion in GB", so no.

File: modules/azure/aks/defender.tf (hunk 1 above)
Comment (landerss1) [reply modules/azure/aks/defender.tf:6]: OK, I think it looks weird to use a for-each on something that is on or off, but sure, I can fix that.

**Review by phillebaba — DISMISSED** [review]
LGTM

**Review by phillebaba — APPROVED** [review]

**Pull request merged**